subcommand/server-acl-init/rules_test.go

```diff
@@ -504,28 +504,51 @@ func TestControllerRules(t *testing.T) {
 	cases := []struct {
 		Name             string
 		EnableNamespaces bool
+		DestConsulNS     string
+		Mirroring        bool
+		MirroringPrefix  string
 		Expected         string
 	}{
 		{
-			"Namespaces are disabled",
-			false,
-			`operator = "write"
-node_prefix "" {
-  policy = "write"
-}
+			Name:             "namespaces=disabled",
+			EnableNamespaces: false,
+			Expected: `operator = "write"
 service_prefix "" {
   policy = "write"
 }`,
 		},
 		{
-			"Namespaces are enabled",
-			true,
-			`operator = "write"
-node_prefix "" {
-  policy = "write"
-}
-service_prefix "" {
-  policy = "write"
+			Name:             "namespaces=enabled, consulDestNS=consul",
+			EnableNamespaces: true,
+			DestConsulNS:     "consul",
+			Expected: `operator = "write"
+namespace "consul" {
+  service_prefix "" {
+    policy = "write"
+  }
+}`,
+		},
+		{
+			Name:             "namespaces=enabled, mirroring=true",
+			EnableNamespaces: true,
+			Mirroring:        true,
+			Expected: `operator = "write"
+namespace_prefix "" {
+  service_prefix "" {
+    policy = "write"
+  }
+}`,
+		},
+		{
+			Name:             "namespaces=enabled, mirroring=true, mirroringPrefix=prefix-",
+			EnableNamespaces: true,
+			Mirroring:        true,
```

Comment:
This is more of a question than a comment on the PR. Do we have a validation somewhere that fails if `mirroring` is `true` but `EnableNamespaces` is `false?

Reply:
No, that's a good idea though. I'm guessing that's best done in the helm chart? It could fail out?

Reply:
It is inconsistent to only do it here though... I'll change to be like the rest and add a note on our backlog.